src/Native/System.Net.Security.Native/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+project(System.Net.Security.Native)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+add_definitions(-DPIC=1)
+
+if (HAVE_HEIMNTLM_HEADERS)
+   find_library(LIB_HEIMNTLM NAMES heimntlm PATHS /usr/heimdal/lib)
+   include_directories(/usr/heimdal/include)
+
+   if(LIB_HEIMNTLM STREQUAL LIB_HEIMNTLM-NOTFOUND)
+       message(FATAL_ERROR "Cannot find libheimntlm.so and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+   endif()
+
+
+   set(NATIVENTLM_SOURCES
+     pal_ntlmapi.cpp
+   )
+
+   add_library(System.Net.Security.Native
+    SHARED
+    ${NATIVENTLM_SOURCES}
+   )
+
+   target_link_libraries(System.Net.Security.Native
+           ${LIB_HEIMNTLM}
+           ${OPENSSL_CRYPTO_LIBRARY}
+   )
+
+   install (TARGETS System.Net.Security.Native DESTINATION .)
+
+elseif (CMAKE_SYSTEM_NAME STREQUAL Linux)
+   message (FATAL_ERROR "cannot find libheim ntlm headers and System.Net.Security.Native cannot build without it. Try installing libheimntlm (or the appropriate package for your platform)")
+else ()
```

Comment:
> Interpretation 1

That's my understanding from @vijaykota and @kapilash.

> In that case, how do we accomplish:

I'm proposing we have the native build emit a warning, and we have the tests skip if the entry points aren't available.  We could then pass a trait or something that says "no, really, you must run this test" such that the CI system could pass that trait and we'd know they were running and either passing or failing.

> If "short term" means "99.97% likely to be solved before RTM"

That's my hope.  @vijaykota and @kapilash will need to comment.  I've already asked for further investigations into other options, including using a completely different dependency, working with whomever to get the packages fixed, etc.

> Otherwise something like (2)

Option (2) does produce a nicer exception, e.g. a PlatformNotSupportedException instead of a DllNotFoundException.  But how does that address your other concerns?